### DISCUSSION TITLE
Tests fail when adding some files to ADDITIONAL_APP_OBJECTS in the Makefile

### DISCUSSION THREAD

**Discussion by thiloschmid** [idaholab/moose#19267] [Q&A Getting Started] [answered]
Dear moose community, 

I am running into an issue when trying to couple an external framework to moose. The additional files are added in the `Makefile` via

    # object files
    ADDITIONAL_APP_OBJECTS += $(patsubst %.cpp, %.$(obj-suffix), $(additional_srcfiles))
    
    # dependencies 
    ADDITIONAL_APP_DEPS += $(patsubst %.cpp, %.$(obj-suffix).d, $(additional_srcfiles))

But after running `make`, the test fails with the following output. 

    (moose) me@cluster:~/Projects/chamois$ ./run_tests 
    <frozen importlib._bootstrap>:228: RuntimeWarning: compiletime version 3.8 of module 'hit' does not match runtime version 3.9
    test:kernels/simple_diffusion.test: Working Directory: /local/home/schmithi/Projects/chamois/test/tests/kernels/simple_diffusion
    test:kernels/simple_diffusion.test: Running command: /local/home/schmithi/Projects/chamois/chamois-opt -i simple_diffusion.i --no-gdb-backtrace
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: Framework Information:
    test:kernels/simple_diffusion.test: MOOSE Version:           git commit 4aee48977f on 2021-10-30
    test:kernels/simple_diffusion.test: LibMesh Version:         
    test:kernels/simple_diffusion.test: PETSc Version:           3.15.1
    test:kernels/simple_diffusion.test: SLEPc Version:           3.15.1
    test:kernels/simple_diffusion.test: Current Time:            Sat Oct 30 14:57:38 2021
    test:kernels/simple_diffusion.test: Executable Timestamp:    Sat Oct 30 14:57:36 2021
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: Parallelism:
    test:kernels/simple_diffusion.test:   Num Processors:          1
    test:kernels/simple_diffusion.test:   Num Threads:             1
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: Mesh: 
    test:kernels/simple_diffusion.test:   Parallel Type:           replicated
    test:kernels/simple_diffusion.test:   Mesh Dimension:          2
    test:kernels/simple_diffusion.test:   Spatial Dimension:       2
    test:kernels/simple_diffusion.test:   Nodes:                   121
    test:kernels/simple_diffusion.test:   Elems:                   100
    test:kernels/simple_diffusion.test:   Num Subdomains:          1
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: Nonlinear System:
    test:kernels/simple_diffusion.test:   Num DOFs:                121
    test:kernels/simple_diffusion.test:   Num Local DOFs:          121
    test:kernels/simple_diffusion.test:   Variables:               "u" 
    test:kernels/simple_diffusion.test:   Finite Element Types:    "LAGRANGE" 
    test:kernels/simple_diffusion.test:   Approximation Orders:    "FIRST" 
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: Execution Information:
    test:kernels/simple_diffusion.test:   Executioner:             Steady
    test:kernels/simple_diffusion.test:   Solver Mode:             Preconditioned JFNK
    test:kernels/simple_diffusion.test:   PETSc Preconditioner:    hypre boomeramg 
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test:  0 Nonlinear |R| = 3.082207e+00
    test:kernels/simple_diffusion.test:       0 Linear |R| = 3.082207e+00
    test:kernels/simple_diffusion.test:       1 Linear |R| = 1.034260e-01
    test:kernels/simple_diffusion.test:       2 Linear |R| = 2.616261e-03
    test:kernels/simple_diffusion.test:       3 Linear |R| = 9.194723e-05
    test:kernels/simple_diffusion.test:       4 Linear |R| = 1.366911e-06
    test:kernels/simple_diffusion.test:  1 Nonlinear |R| = 1.373820e-06
    test:kernels/simple_diffusion.test:       0 Linear |R| = 1.373820e-06
    test:kernels/simple_diffusion.test:       1 Linear |R| = 3.507042e-08
    test:kernels/simple_diffusion.test:       2 Linear |R| = 4.049791e-10
    test:kernels/simple_diffusion.test:       3 Linear |R| = 1.351613e-11
    test:kernels/simple_diffusion.test:  2 Nonlinear |R| = 1.352021e-11
    test:kernels/simple_diffusion.test:  Solve Converged!
    test:kernels/simple_diffusion.test: double free or corruption (!prev)
    test:kernels/simple_diffusion.test: Aborted
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: Framework Information:
    test:kernels/simple_diffusion.test: MOOSE Version:           git commit 4aee48977f on 2021-10-30
    test:kernels/simple_diffusion.test: LibMesh Version:         
    test:kernels/simple_diffusion.test: PETSc Version:           3.15.1
    test:kernels/simple_diffusion.test: SLEPc Version:           3.15.1
    test:kernels/simple_diffusion.test: Current Time:            Sat Oct 30 14:57:38 2021
    test:kernels/simple_diffusion.test: Executable Timestamp:    Sat Oct 30 14:57:36 2021
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: Parallelism:
    test:kernels/simple_diffusion.test:   Num Processors:          1
    test:kernels/simple_diffusion.test:   Num Threads:             1
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: Mesh: 
    test:kernels/simple_diffusion.test:   Parallel Type:           replicated
    test:kernels/simple_diffusion.test:   Mesh Dimension:          2
    test:kernels/simple_diffusion.test:   Spatial Dimension:       2
    test:kernels/simple_diffusion.test:   Nodes:                   121
    test:kernels/simple_diffusion.test:   Elems:                   100
    test:kernels/simple_diffusion.test:   Num Subdomains:          1
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: Nonlinear System:
    test:kernels/simple_diffusion.test:   Num DOFs:                121
    test:kernels/simple_diffusion.test:   Num Local DOFs:          121
    test:kernels/simple_diffusion.test:   Variables:               "u" 
    test:kernels/simple_diffusion.test:   Finite Element Types:    "LAGRANGE" 
    test:kernels/simple_diffusion.test:   Approximation Orders:    "FIRST" 
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: Execution Information:
    test:kernels/simple_diffusion.test:   Executioner:             Steady
    test:kernels/simple_diffusion.test:   Solver Mode:             Preconditioned JFNK
    test:kernels/simple_diffusion.test:   PETSc Preconditioner:    hypre boomeramg 
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test:  0 Nonlinear |R| = 3.082207e+00
    test:kernels/simple_diffusion.test:       0 Linear |R| = 3.082207e+00
    test:kernels/simple_diffusion.test:       1 Linear |R| = 1.034260e-01
    test:kernels/simple_diffusion.test:       2 Linear |R| = 2.616261e-03
    test:kernels/simple_diffusion.test:       3 Linear |R| = 9.194723e-05
    test:kernels/simple_diffusion.test:       4 Linear |R| = 1.366911e-06
    test:kernels/simple_diffusion.test:  1 Nonlinear |R| = 1.373820e-06
    test:kernels/simple_diffusion.test:       0 Linear |R| = 1.373820e-06
    test:kernels/simple_diffusion.test:       1 Linear |R| = 3.507042e-08
    test:kernels/simple_diffusion.test:       2 Linear |R| = 4.049791e-10
    test:kernels/simple_diffusion.test:       3 Linear |R| = 1.351613e-11
    test:kernels/simple_diffusion.test:  2 Nonlinear |R| = 1.352021e-11
    test:kernels/simple_diffusion.test:  Solve Converged!
    test:kernels/simple_diffusion.test: double free or corruption (!prev)
    test:kernels/simple_diffusion.test: Aborted
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test: Exit Code: 134
    test:kernels/simple_diffusion.test: ################################################################################
    test:kernels/simple_diffusion.test: Tester failed, reason: CRASH
    test:kernels/simple_diffusion.test: 
    test:kernels/simple_diffusion.test ............................................................ FAILED (CRASH)
    
    
    Final Test Results:
    --------------------------------------------------------------------------------------------------------------
    test:kernels/simple_diffusion.test ............................................................ FAILED (CRASH)
    --------------------------------------------------------------------------------------------------------------
    Ran 1 tests in 0.2 seconds. Average test time 0.1 seconds, maximum test time 0.1 seconds.
    0 passed, 0 skipped, 0 pending, 1 FAILED

I am not too experienced with building more complex projects. Other input files lead to a similar result. They almost complete and lead to this `double free or corruption (!prev)` error somewhere after the solver converges. I have not been able to track down the problem with the debugger since the stack is corrupted. 

It seems weird to me that the simple diffusion test fails even though it should not depend on the third-party library. Is there some moose function that is overloaded in some unintended way? 

I would appreciate any help toward resolving this issue.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Can you please build in debug mode and get us a backtrace in a debugger?
instructions are here: https://mooseframework.inl.gov/application_development/debugging.html#!

Guillaume

**Reply by thiloschmid** [reply]
I was able to resolve the issue. My apologies, it had nothing to do with moose directly.
